articles/modules/ROOT/pages/using-subqueries-to-avoid-the-eager.adoc

```diff
@@ -0,0 +1,222 @@
+
+With these changes, per id from the UNWIND, the subquery will execute in full. Here is what would result if we actually ran the query:
+
+* The first row from the UNWIND, id 1, will start subquery execution.
+* It will execute the MATCH, and find the existing node with id: 1.
+* MERGE will execute, creating a node with id:2.
+* The RETURNs will execute, ending the subquery execution for that row, and producing the first output row from the query.
+* The second row from the UNWIND, id 2, will start subquery execution.
+* It will execute the MATCH, and find the just-created node with id: 2.
+* MERGE will execute, creating a node with id:3.
+* The RETURNs will execute, ending the subquery execution for that row, and producing the second output row from the query.
+* The subsequent rows from the UNWIND will execute in a similar manner from the subquery.
+* As a result of the subquery scoping the Eager and enforcing per-row execution here, a single run of this query will produce 5 new nodes, for a total of 6, with ids of 1 through 6.
+
+While this has changed behavior such that it won't pressure the heap, and will again allow sane batch processing, it is important to note that the query results changed!
+
+Again, this is because subqueries enforce per-row processing behavior, but only between the input rows at the point of the subquery call, and the entirety of the single subquery.
+If two separate back-to-back subquery calls were used instead, then the planner could possibly plan an Eager between those calls, and you would need to check to see if that meets your expectations for the behavior and results you want, or needs to be further tuned to remove the Eager.
+
+=== Scoping the Eager
+
+Even though the subquery usage has changed the behavior, and allowed us to process in a per-row manner, the Eager is still in the query plan:
+
+image::https://i.imgur.com/HwfyuU6.jpeg[]
+
+The difference is in where the Eager occurs in the plan.
+In this case, the Eager is on the right-hand side of an Apply operator.
+
+The Apply operator means: for each input row from the left side, do all the stuff on the right side of the operator.
+Here's the official docs for Apply:
+
+https://neo4j.com/docs/cypher-manual/4.4/execution-plans/operators/#query-plan-apply
+
+Subqueries generate Apply operators, so this plan just confirms that the Eager is scoped to an individual subquery execution, and won't alter behavior outside of the subquery.
+
+When managing eager behavior, this kind of plan is what you're looking for to confirm that the Eager is scoped behind an Apply,
+and not on the main branch of execution, which is the direct line of operators from the top-leftmost operator (which may not be at the top of the plan, so check carefully) to the last operator at the bottom.
+
+== Nested subqueries for additional scoping
+
+For a more complex query, a single subquery usage may not be enough to properly reign in the eager behavior.
```

Comment:
this feels like the hacks people do with cache-line padding where they create one subclass with one pad to avoid the CPU/JVM reordering fields